### PR TITLE
Of11 libraries

### PR DESCRIPTION
**Pull request by colemanjs**
This PR improves the organization of additivefoam libraries and moves foamToExaCA, which currently doesn't work for dynamic meshes, to a FunctionObject named ExaCA

### CODE REVIEW

File: applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+bool Foam::functionObjects::ExaCA::write()
+{
+    return true;
+}
+
+
+void Foam::functionObjects::ExaCA::movePoints(const polyMesh& mesh)
+{
+    if (&mesh == &mesh_)
```
[the remark this answers is not in the document]
Comment (colemanjs) [reply applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C:136]: This was borrowed from other FunctionObjects for sampling (https://github.com/OpenFOAM/OpenFOAM-11/blob/c7c47ff3f7b39ba0240b9bf5753c7e86bddaeb92/src/sampling/sampledSet/sampledSets/sampledSets.C#L280)

You can see function descriptions here:
https://github.com/OpenFOAM/OpenFOAM-11/blob/c7c47ff3f7b39ba0240b9bf5753c7e86bddaeb92/src/sampling/sampledSet/sampledSets/sampledSets.H#L175C1-L185C61

File: applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+bool Foam::functionObjects::ExaCA::execute()
+{
+    if (mesh_.time().writeTime())
+    {
+        pointScalarField Tp("Tp", vpi_.interpolate(T_));
+        Tp.write();
+    }
+
+    //Info << gMax(T_()) << endl;
```
[the remark this answers is not in the document]
Comment (colemanjs) [reply applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C]: That was a mistake. Corrected in #25

File: tutorials/AMB2018-02-B/system/ExaCA [hunk 3]
```diff
@@ -10,32 +10,6 @@ ExaCA
...
-
-    interpolationScheme pointMVC;
-
-    sets
-    (
-        data
-        {
-            type        boxUniform;
-            axis        xyz;
-            box         (0 -0.0001 -0.0002) (0.002 0.0001 0);
```
Comment (kincaidkc) [inline tutorials/AMB2018-02-B/system/ExaCA:31]: where is the bounding box information kept in the new ExaCA FunctionObject?

Comment (kincaidkc) [reply tutorials/AMB2018-02-B/system/ExaCA:31]: or is it no longer needed? is matt doing that part of the conversion in ExaCA itself?

File: applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+
+Foam::wordList Foam::functionObjects::ExaCA::fields() const
+{
+    return wordList::null();
+}
+
+
+bool Foam::functionObjects::ExaCA::execute()
+{
+    if (mesh_.time().writeTime())
```
Comment (kincaidkc) [inline applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C:111]: are you sure you want to tie the `ExaCA` write to the main write? the function objects class has its own execute/write controls and some users might want `ExaCA` data more often than the other fields

File: tutorials/AMB2018-02-B/system/ExaCA (hunk 3 above)
Comment (colemanjs) [reply tutorials/AMB2018-02-B/system/ExaCA:31]: The ExaCA FunctionObject is just returning the temperature at vertices at the moment. I will need to work on the logic to actually interpolate and write the data to exaca points. This comment was removed in [fbba891](https://github.com/kincaidkc/AdditiveFOAM/pull/25/commits/fbba891ccd5b5200279a1453f7588b0c9fb7c7eb)

File: applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C (hunk 4 above)
Comment (colemanjs) [reply applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C:111]: Yeah, this was just a verification exercise that the volPointInterpolation weights were bring properly updated with AMR. The ExaCA FunctionObject isn't doing anything useful at the moment, other than getting the code structure in place

Comment (kincaidkc) [reply applications/solvers/additiveFoam/FunctionObjects/ExaCA/ExaCA.C:111]: ok cool, thanks for the clarification

**Pull request merged**